includes/utils/class-amp-image-dimension-extractor.php

```diff
@@ -2,20 +2,44 @@
 
 class AMP_Image_Dimension_Extractor {
 	static $callbacks_registered = false;
+    const STATUS_FAILED_LAST_ATTEMPT = 'failed';
+    const STATUS_IMAGE_EXTRACTION_FAILED= 'failed';
 
-	static public function extract( $url ) {
+    static public function extract( $urls ) {
```

Comment:
Right, so it's unlikely that people are calling `extract` directly, but they would be hooking into `amp_extract_image_dimensions` ([we're doing it ourselves on WP.com](https://github.com/Automattic/amp-wp/blob/master/wpcom-helper.php#L106)) which is now changed from two parameters (`bool|array $dimensions, string $url`) to one param (`array $urls`). That's definitely going to break stuff.

```
-		return apply_filters( 'amp_extract_image_dimensions', false, $url );
+		$dimensions = apply_filters( 'amp_extract_image_dimensions', $urls );
```

So yeah, we may need a new method + filter (since the existing method+filter didn't really plan for parallel/batch processing).